lib/internal/test_runner/tap_parser.js

```diff
@@ -548,8 +548,24 @@ class TapParser extends Transform {
       switch (kind) {
         case TokenKind.TAP:
           return this.#Version();
-        case TokenKind.NUMERIC:
-          return this.#Plan();
+        case TokenKind.NUMERIC: {
+          const planStart = this.#next();
+          if (planStart.kind !== TokenKind.NUMERIC) {
+            this.#error('Expected a plan start count');
+            break;
+          }
+          const planToken = this.#next();
+          if (planToken?.kind !== TokenKind.TAP_PLAN) {
+            this.#error('Expected ".." symbol');
+            break;
+          }
+          const planEnd = this.#next();
+          if (!planEnd || planEnd?.kind !== TokenKind.NUMERIC) {
+            this.#error('Expected a plan end count');
+            break;
+          }
```

Comment:
In order to keep the code more maintainable and readable, please move these changes to #Plan().

Reply:
Hey, @manekinekko.
I totally agree that moving this logic inside #Plan() is good,  but how would we go about [doing this](https://github.com/nodejs/node/pull/46056#pullrequestreview-1240503081)?
I'm not sure, but using a try-catch block could be a nice idea. Could you please assist me?

Reply:
See https://github.com/nodejs/node/pull/46056#issuecomment-1410606209